core/src/main/java/io/questdb/network/AbstractIODispatcher.java

```diff
@@ -213,6 +213,14 @@ public boolean processIOQueue(IORequestProcessor<C> processor) {
             C connectionContext = event.context;
             final int operation = event.operation;
             ioEventSubSeq.done(cursor);
+            try {
+                connectionContext.init();
+            } catch (CairoException e) {
+                LOG.error().$("could not initialize connection context [fd=").$(connectionContext.getFd())
```

Comment:
this is likely not tested - when context fails to initialize we must not pass it on into onRequest() - best course of action is to disconnect, because at this point we do not know what the protocol is.

Reply:
here context.isInitialized() not exposed, how is this not random what onRequest() will walk into ?